lib/src/network_util.dart

```diff
@@ -23,20 +23,35 @@ class NetworkUtil {
         List<dynamic> routeList = parsedJson["routes"];
         for (var route in routeList) {
           results.add(PolylineResult(
-              points:
-                  PolylineDecoder.run(route["overview_polyline"]["points"]),
-              errorMessage: "",
-              status: parsedJson["status"],
-              distance: route["legs"][0]["distance"]["text"],
-              distanceValue: route["legs"][0]["distance"]["value"],
-              overviewPolyline: route["overview_polyline"]["points"],
-              durationValue: route["legs"][0]["duration"]["value"],
-              endAddress: route["legs"][0]['end_address'],
-              startAddress: route["legs"][0]['start_address'],
-              duration: route["legs"][0]["duration"]["text"]));
+            points: PolylineDecoder.run(route["overview_polyline"]["points"]),
+            errorMessage: "",
+            status: parsedJson["status"],
+            totalDistanceValue: route['legs']
+                .map((leg) => leg['distance']['value'])
+                .reduce((v1, v2) => v1 + v2),
+            distanceTexts: <String>[
+              ...route['legs'].map((leg) => leg['distance']['text'])
+            ],
+            distanceValues: <int>[
+              ...route['legs'].map((leg) => leg['distance']['value'])
+            ],
+            overviewPolyline: route["overview_polyline"]["points"],
+            totalDurationValue: route['legs']
+                .map((leg) => leg['duration']['value'])
+                .reduce((v1, v2) => v1 + v2),
+            durationTexts: <String>[
+              ...route['legs'].map((leg) => leg['duration']['text'])
+            ],
+            durationValues: <int>[
+              ...route['legs'].map((leg) => leg['distance']['value'])
```

Comment:
Little mistake, this is duration and not distance here: 
```suggestion
              ...route['legs'].map((leg) => leg['duration']['value'])
```